opm/simulators/linalg/PreconditionerFactory_impl.hpp

```diff
@@ -404,115 +471,129 @@ struct StandardPreconditioners<Operator,Dune::Amg::SequentialInformation>
                 const std::string smoother = prm.get<std::string>("smoother", "ParOverILU0");
                 if (smoother == "ILU0" || smoother == "ParOverILU0") {
                     using Smoother = SeqILU<M, V, V>;
-                    return AMGHelper<O,C,M,V>::template makeAmgPreconditioner<Smoother>(op, prm, true);
+                    return AMGHelper<O, C, M, V>::template makeAmgPreconditioner<Smoother>(op, prm, true);
                 } else if (smoother == "Jac") {
                     using Smoother = SeqJac<M, V, V>;
-                    return AMGHelper<O,C,M,V>::template makeAmgPreconditioner<Smoother>(op, prm, true);
+                    return AMGHelper<O, C, M, V>::template makeAmgPreconditioner<Smoother>(op, prm, true);
                 } else if (smoother == "SOR") {
                     using Smoother = SeqSOR<M, V, V>;
-                    return AMGHelper<O,C,M,V>::template makeAmgPreconditioner<Smoother>(op, prm, true);
-                    // } else if (smoother == "GS") {
-                    //     using Smoother = SeqGS<M, V, V>;
-                    //     return makeAmgPreconditioner<Smoother>(op, prm, true);
+                    return AMGHelper<O, C, M, V>::template makeAmgPreconditioner<Smoother>(op, prm, true);
+                } else if (smoother == "GS") {
+                    using Smoother = SeqGS<M, V, V>;
+                    return AMGHelper<O, C, M, V>::template makeAmgPreconditioner<Smoother>(op, prm, true);
                 } else if (smoother == "SSOR") {
                     using Smoother = SeqSSOR<M, V, V>;
-                    return AMGHelper<O,C,M,V>::template makeAmgPreconditioner<Smoother>(op, prm, true);
+                    return AMGHelper<O, C, M, V>::template makeAmgPreconditioner<Smoother>(op, prm, true);
                 } else if (smoother == "ILUn") {
                     using Smoother = SeqILU<M, V, V>;
-                    return AMGHelper<O,C,M,V>::template makeAmgPreconditioner<Smoother>(op, prm, true);
+                    return AMGHelper<O, C, M, V>::template makeAmgPreconditioner<Smoother>(op, prm, true);
                 } else {
-                    OPM_THROW(std::invalid_argument,
-                              "Properties: No smoother with name " + smoother + ".");
+                    OPM_THROW(std::invalid_argument, "Properties: No smoother with name " + smoother + ".");
                 }
             });
             F::addCreator("famg", [](const O& op, const P& prm, const std::function<V()>&, std::size_t) {
-                auto crit = AMGHelper<O,C,M,V>::criterion(prm);
+                auto crit = AMGHelper<O, C, M, V>::criterion(prm);
                 Dune::Amg::Parameters parms;
                 parms.setNoPreSmoothSteps(1);
                 parms.setNoPostSmoothSteps(1);
-                return wrapPreconditioner<Dune::Amg::FastAMG<O, V>>(op, crit, parms);
+                return getDummyUpdateWrapper<Dune::Amg::FastAMG<O, V>>(op, crit, parms);
```

Comment:
The FastAMG should probably not use the dummy wrapper. Changing this to the new rebuild-on-update wrapper would be a bugfix I think. Although the interface is not the same, so it would probably not compile...

Postponing this fix until we have a more generic rebuild wrapper.

Reply:
Yes, this will be more easily fixed when we have the more generic rebuildOnUpdate class